typescript/processors/readTypeScriptModules.js

```diff
@@ -0,0 +1,451 @@
+var glob = require('glob');
+var path = require('canonical-path');
+var _ = require('lodash');
+var ts = require('typescript');
+
+module.exports = function readTypeScriptModules(tsParser, modules, getFileInfo, untouchedNamespaces,
+                                                getExportDocType, getContent, createDocMessage, log) {
+
```

Comment:
@davidreher I'm on the latest commit e423fb7, but I still have the following error for the setup above:

```sh
error:   Invalid property in "readFilesProcessor" (in "base" package)
error:    basePath=[Base path can't be blank], sourceFiles=[Source files can't be blank], fileReaders=[File readers can't be blank]
error:   Error processing docs:  processor=readFilesProcessor, package=base, basePath=[Base path can't be blank], sourceFiles=[Source files can't be blank], fileReaders=[File readers can't be blank]
```

After faking the ```readFilesProcessor``` as @coldacid suggested, it steps through into the readTypeScriptModules, but my output index files only contains the header (which is the path to the file) and not the members.

The log says in the ```writeFilesProcessor``` section:

```sh
debug:   Document "/Users/myusr/dev/dgeni-example/src/logger/Logger.log, member" has no outputPath.
```

which origins from a prior message in the ```computePathsProcessor``` section:

```sh
debug:   path: //Users/myusr/dev/dgeni-example/src/logger/Logger#log; outputPath: undefined - doc "/Users/myusr/dev/dgeni-example/src/logger/Logger.log" (member)  - from file "/Users/myusr/dev/dgeni-example/src/logger.ts"
```

I have push my typescript sample project here: [pvadam/dgeni-example#typescript-example](https://github.com/pvadam/dgeni-example/tree/typescript-example).